packages/noya-state/src/reducers/textStyleReducer.ts

```diff
@@ -178,13 +201,23 @@ export function textStyleReducer(
 
         return produce(state, (draft) => {
           accessPageLayers(draft, pageIndex, layerIndexPaths).forEach(
-            (layer) => {
+            (layer, index) => {
               if (!Layers.isTextLayer(layer) || !Layers.hasTextStyle(layer))
                 return;
 
               const attributes = layer.style.textStyle.encodedAttributes;
 
               attributes.MSAttributedStringTextTransformAttribute = action[1];
+
+              if (index === 0) {
```

Comment:
We do this check for `index === 0` in a few places, and I think it should work, but it's a little unclear what it's for. I think it might be clearer if we do that mutation outside the loop. The also solves the potential issue of e.g. a rectangle as the first selected layer, which would cause the `index === 0` check to not work.

```ts
return produce(state, (draft) => {
   const textLayers = accessPageLayers(draft, pageIndex, layerIndexPaths).filter(Layers.isTextLayer)
  
   textLayers.forEach(...)

   const firstTextLayer = textLayers[0]

   if (firstTextLayer) {
      draft.lastEditedTextStyle = ...
   }
 })
```